sdk/trace/src/main/java/io/opentelemetry/sdk/trace/samplers/SamplingResult.java

```diff
@@ -23,48 +23,112 @@
 public interface SamplingResult {
 
   /**
-   * Returns a {@link SamplingResult} with no attributes and {@link SamplingResult#getDecision()}
-   * returning {@code decision}.
+   * Returns a {@link SamplingResult} corresponding to {@link SamplingDecision#RECORD_AND_SAMPLE}
+   * with no attributes and {@link SamplingResult#getDecision()} returning {@code decision}.
+   *
+   * <p>Use {@link #recordAndSample(Attributes)} if you need attributes.
+   *
+   * <p>This is meant for use by custom {@link Sampler} implementations.
+   *
+   * @return A {@link SamplingResult} with empty attributes and the provided {@code decision}.
+   */
+  static SamplingResult recordAndSample() {
+    return ImmutableSamplingResult.getEmptyRecordedAndSampledSamplingResult();
+  }
+
+  /**
+   * Returns a {@link SamplingResult} corresponding to {@link SamplingDecision#RECORD_AND_SAMPLE}
+   * with the given {@code attributes} and {@link SamplingResult#getDecision()} returning
+   * {@code decision}.
+   *
+   * <p>This is meant for use by custom {@link Sampler} implementations.
+   *
+   * <p>Using {@link #recordAndSample()} instead of this method is slightly faster and
+   * shorter if you don't need attributes.
+   *
+   * @param attributes The attributes to return from {@link SamplingResult#getAttributes()}. A
+   *     different object instance with the same elements may be returned.
+   * @return A {@link SamplingResult} with the attributes equivalent to {@code attributes} and the
+   * {@link SamplingDecision#RECORD_AND_SAMPLE}
+   */
+  static SamplingResult recordAndSample(Attributes attributes) {
```

Comment:
I wouldn't create these overloads with attributes. As @anuraaga mentioned, the existing method works fine for that, and actually does create a new object when being called.

Reply:
Thanks, but in the project, `RateLimitingSampler` calls the method with parameters. I think the method with parameters is necessary when we need specific sampling. I'm a first-time contributor, I hope you can give me some guidance.

Reply:
@rocksnake Thanks for the help - as per the below comment, we won't be (can't) removing the `create` methods. We should continue to use `create` for creating the results with attributes

Reply:
Ok, thanks. And in my last commit, I have restored the original create method.